src/workerd/api/sockets.c++

```diff
@@ -41,6 +41,20 @@ bool isValidHost(kj::StringPtr host) {
   return true;
 }
 
+SecureTransportKind parseSecureTransport(SocketOptions* opts) {
+  auto value = KJ_UNWRAP_OR_RETURN(opts->secureTransport, SecureTransportKind::OFF).begin();
```

Comment:
I'm not up to date one the spec here, if defaulting to `OFF` is correct then this PR is ready to go.